airflow/models.py

```diff
@@ -3368,6 +3416,37 @@ def __repr__(self):
     def id_for_date(klass, date, prefix=ID_FORMAT_PREFIX):
         return prefix.format(date.isoformat()[:19])
 
+    @provide_session
+    def refresh_from_db(self, session=None):
+        DR = DagRun
+
+        dr = session.query(DR).filter(
+            DR.dag_id == self.dag_id,
+            DR.execution_date == self.execution_date,
+            DR.run_id == self.run_id
+        ).first()
+        if dr:
+            self.id = dr.id
+            self.state = dr.state
+
+    @staticmethod
+    def find(dag_id, run_id=None, state=None, external_trigger=None):
+        DR = DagRun
+
+        session = settings.Session()
```

Comment:
Fair point. I think we should be consistent throughout and go with one way or another but that is out of scope.